examples/src/main/scala/TracingExample.scala

```diff
@@ -21,24 +21,35 @@ import cats.effect.Resource
 import cats.effect.std.Console
 import cats.syntax.all._
 import io.opentelemetry.api.GlobalOpenTelemetry
+import org.typelevel.otel4s.Otel4s
+import org.typelevel.otel4s.TextMapPropagator
 import org.typelevel.otel4s.java.OtelJava
+import org.typelevel.otel4s.trace.SpanContext
 import org.typelevel.otel4s.trace.Tracer
+import org.typelevel.vault.Vault
 
 import scala.concurrent.duration._
 
 trait Work[F[_]] {
-  def doWork: F[Unit]
+  def request(headers: Map[String, String]): F[Unit]
 }
 
 object Work {
-  def apply[F[_]: Monad: Tracer: Console]: Work[F] =
+  def apply[F[_]: Monad: Tracer: TextMapPropagator: Console]: Work[F] =
     new Work[F] {
-      def doWork: F[Unit] =
-        Tracer[F].span("Work.DoWork").use { span =>
-          span.addEvent("Starting the work.") *>
-            doWorkInternal *>
-            span.addEvent("Finished working.")
+      def request(headers: Map[String, String]): F[Unit] = {
+        val vault =
+          implicitly[TextMapPropagator[F]].extract(Vault.empty, headers)
+        Tracer[F].childOrContinue(SpanContext.fromContext(vault)) {
+          Tracer[F].span("Work.DoWork").use { span =>
+            Tracer[F].currentSpanContext
+              .flatMap(ctx => Console[F].println("Context is " + ctx)) *>
+              span.addEvent("Starting the work.") *>
+              doWorkInternal *>
+              span.addEvent("Finished working.")
+          }
         }
```

Comment:
```suggestion
        Tracer[F].currentSpanContext.flatMap { current =>
          Tracer[F].childOrContinue(SpanContext.fromContext(vault)) {
            val builder = Tracer[F].spanBuilder("Work.DoWork")
            current.fold(builder)(builder.addLink(_)).build.use { span =>
              Tracer[F].currentSpanContext
                .flatMap(ctx => Console[F].println("Context is " + ctx)) *>
                span.addEvent("Starting the work.") *>
                doWorkInternal *>
                span.addEvent("Finished working.")
            }
          }
        }
```

We can add a link to the parent (resource) span. Otherwise, there is no way to associate spans.

In Jaeger it will look like this:

<img width="852" alt="image" src="https://user-images.githubusercontent.com/6395483/219600338-c371d26e-7f00-4459-be5d-db11d433932a.png">

<img width="738" alt="image" src="https://user-images.githubusercontent.com/6395483/219600402-5b535ff6-0343-4d84-a3a3-d26b89e4ba22.png">